pkg/aggregator/demultiplexer.go

```diff
@@ -643,6 +643,7 @@ func sendIterableSeries(serializer serializer.MetricSerializer, start time.Time,
 	err := serializer.SendIterableSeries(series)
 	// if err == nil, SenderStopped was called and it is safe to read the number of series.
 	count := series.SeriesCount()
+	series.IterationStopped()
```

Comment:
It looks that this is always called as soon as `SendIterableSeries` returns. Is there a situation when this wouldn't be true? Would it make sense to move this call into `SendIterableSeries` to make sure it is always called?